ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -885,6 +885,141 @@ public void testExpiryAfterMetadataExchange() throws Exception {
             .collect(Collectors.toSet()));
   }
 
+  /**
+   * Test the case where 1.remote host has different versions of blobIds and local host is empty; 2. remote and local hosts
+   * have different conversion maps (StoreKeyConverter).
+   * The replication between remote and local hosts should honor conversion rules on each node and the final results on
+   * local hosts are as expected.
+   */
+  @Test
+  public void testReplicationWithCombinedVersionsOfBlobIds() throws Exception {
+    MockClusterMap clusterMap = new MockClusterMap();
+    Pair<Host, Host> localAndRemoteHosts = getLocalAndRemoteHosts(clusterMap);
+    Host localHost = localAndRemoteHosts.getFirst();
+    Host remoteHost = localAndRemoteHosts.getSecond();
+
+    MockStoreKeyConverterFactory storeKeyConverterFactory = new MockStoreKeyConverterFactory(null, null);
+    storeKeyConverterFactory.setConversionMap(new HashMap<>());
+    storeKeyConverterFactory.setReturnInputIfAbsent(true);
+    MockStoreKeyConverterFactory.MockStoreKeyConverter storeKeyConverter =
+        storeKeyConverterFactory.getStoreKeyConverter();
+
+    PartitionId partitionId = clusterMap.getWritablePartitionIds(null).get(0);
+    // Only Version_2 and Version_5 blobIds are used in this tests.
+    // For simplicity, we use A1 to denote Version_2 blobId, A2 to denote Version_5 blobId.
+    // For example, A1P means Version_2 PUT message, A2D means Version_5 DELETE message, etc.
```

Comment:
Instead of A1 and A2 use O (for old) and N (for new)

Reply:
Right, O(old) and N(new) may be more appropriate for open source code. Fixed.